Lib/test/test_pyrepl/test_pyrepl.py

```diff
@@ -31,6 +31,14 @@
 except ImportError:
     pty = None
 
+
+clean_env = os.environ.copy()
+# cleanup from PYTHON* variables
```

Comment:
This should not be changed at module level, but this functionality should be added to `self.run_repl` so it uses `-E` in the subprocess call or instead of using a subprocess it runs `test.support.run_python`

Reply:
@pablogsal, not sure I get your point.  Something along this is doable in the ``run_repl()``, yes.  But this require adding some option(s?) to whitelist some ``PYTHON*`` (like ``PYTHON_HISTORY``) variables, which we *want* to pass to subprocess.

How about using mock.patch to filter out ``os.environ`` in setUp() method?